common/src/main/java/org/apache/celeborn/common/network/server/memory/MemoryManager.java

```diff
@@ -323,7 +331,12 @@ public MemoryManagerStat currentMemoryAction() {
   }
 
   public void trimAllListeners() {
-    memoryPressureListeners.forEach(MemoryPressureListener::onTrim);
+    // First trim data from disk buffer to flusher
+    // Second trim flusher flush data
+    // Finally trim empty netty buffer cache
+    otherListeners.forEach(
```

Comment:
I thinks it's better to sequentially call each listener's ```onTrim``` instead of using a callback